starknet_py/net/client_utils.py

```diff
@@ -17,3 +18,21 @@ def hash_to_felt(value: Hash) -> str:
 
 def is_block_identifier(value: Union[int, Hash, Tag]) -> bool:
     return isinstance(value, str) and value in get_args(Tag)
+
+
+def encode_l1_message(tx: L1HandlerTransaction) -> bytes:
+    # TODO (#1047): remove this assert once GatewayClient is deprecated and nonce is always required
+    assert tx.nonce
```

Comment:
```suggestion
    assert tx.nonce is not None
```

Explicit `None` checks are preferred in starknet.py (at least I was taught that 😄)